### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by srueg**
Closes https://ticket.vshn.net/browse/APPU-2747

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update the ./CHANGELOG.md.
- [x] Link this PR to related issues.

### CODE REVIEW

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+[horizontal]
+type:: dictionary
+default:: Default image registry.
+
+A dictionary holding the `.spec` for the image registry config.
+
+See the https://docs.openshift.com/container-platform/4.4/registry/configuring-registry-operator.html#registry-operator-configuration-resource-overview_configuring-registry-operator[OpenShift docs] for available parameters.
+
+
+=== `config.routes`
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/parameters.adoc:17]:
```suggestion
=== `routes`
```

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+[horizontal]
+type:: array
+default:: undefined
+
+To create custom routes for exposing the registry, create an entry in this array (see <<Example>>).
+The `secretName` can be used to specify a custom TLS certificate.
+By default it uses the default wildcard certificate of the router.
+
+
+=== `config.storage`
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/parameters.adoc:28]:
```suggestion
=== `storage`
```

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+== `config`
+
+[horizontal]
+type:: dictionary
+default:: Default image registry.
+
+A dictionary holding the `.spec` for the image registry config.
+
+See the https://docs.openshift.com/container-platform/4.4/registry/configuring-registry-operator.html#registry-operator-configuration-resource-overview_configuring-registry-operator[OpenShift docs] for available parameters.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/parameters.adoc:14]:
```suggestion
See the https://docs.openshift.com/container-platform/latest/registry/configuring-registry-operator.html#registry-operator-configuration-resource-overview_configuring-registry-operator[OpenShift docs] for available parameters.
```

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+== `pruning`
+
+[horizontal]
+type:: dictionary
+default:: Default pruning options.
+
+A dictionary holding the `.spec` for the image pruning config.
+
+See the https://docs.openshift.com/container-platform/4.4/applications/pruning-objects.html#pruning-images_pruning-objects[OpenShift docs] for available parameters.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/parameters.adoc:48]:
```suggestion
See the https://docs.openshift.com/container-platform/latest/applications/pruning-objects.html#pruning-images_pruning-objects[OpenShift docs] for available parameters.
```

File: component/main.jsonnet [hunk 5]
```diff
@@ -1,10 +1,24 @@
...
 
-// Define outputs below
+local versionGroup = 'imageregistry.operator.openshift.io/v1';
+
 {
+  '00_namespace': kube.Namespace(params.namespace) {
+    metadata+: {
+      annotations:: {},
+      labels+: {
+        'network-policies.syn.tools/no-defaults': 'true',
```
[the remark this answers is not in the document]
Comment (srueg) [reply component/main.jsonnet:13]: Good point. I'd go with 1. just because it's a little bit simpler.

[the remark this answers is not in the document]
Comment (srueg) [reply component/main.jsonnet]: Exposing the labels in https://github.com/projectsyn/component-networkpolicy/pull/2

**Pull request merged**